spot-vaults/contracts/WethWamplManager.sol

```diff
@@ -0,0 +1,325 @@
+        } else {
+            VAULT.setLimitThreshold(INF_LIMIT_RANGE);
+        }
+    }
+
+    /// @dev Trims the vault's current liquidity based on the active liquidity percentage.
+    ///      To be invoked right after a rebalance operation, as it assumes that all of the vault's
+    ///      liquidity has been deployed before trimming.
+    function _trimLiquidity(uint256 activePerc) private {
+        int24 _fullLower = VAULT.fullLower();
+        int24 _fullUpper = VAULT.fullUpper();
+
+        int24 _baseLower = VAULT.baseLower();
+        int24 _baseUpper = VAULT.baseUpper();
+
+        (uint128 fullLiquidity, , , , ) = _position(_fullLower, _fullUpper);
+        (uint128 baseLiquidity, , , , ) = _position(_baseLower, _baseUpper);
+
+        // Recalculated fullLiquidity, baseLiquidity will be lesser than the original value
+        // Thus, there's no risk of overflow.
+        fullLiquidity = uint128(
+            FullMath.mulDivRoundingUp(uint256(fullLiquidity), ONE - activePerc, ONE)
+        );
+        baseLiquidity = uint128(
+            FullMath.mulDivRoundingUp(uint256(baseLiquidity), ONE - activePerc, ONE)
+        );
+
+        VAULT.emergencyBurn(_fullLower, _fullUpper, fullLiquidity);
+        VAULT.emergencyBurn(_baseLower, _baseUpper, baseLiquidity);
+        // NOTE: Limit liquidity remains unchanged here.
+    }
+
+    /// @dev Wrapper around `IUniswapV3Pool.positions()`.
+    function _position(
+        int24 tickLower,
+        int24 tickUpper
+    ) private view returns (uint128, uint256, uint256, uint128, uint128) {
+        bytes32 positionKey = PositionKey.compute(address(this), tickLower, tickUpper);
+        return IUniswapV3Pool(VAULT.pool()).positions(positionKey);
+    }
+
+    /// @dev Fetches most recent report from the given chain link oracle contract.
+    ///      The data is considered invalid if the latest report is stale.
+    ///      The returned report is a fixed point number with {DECIMALS} places.
+    function _getCLOracleData(
+        IChainlinkOracle oracle,
+        uint256 oracleDecimals
+    ) private view returns (uint256, bool) {
+        (, int256 p, , uint256 updatedAt, ) = oracle.latestRoundData();
+        uint256 price = FullMath.mulDiv(uint256(p), ONE, 10 ** oracleDecimals);
+        return (
+            price,
+            (block.timestamp - updatedAt) <= CL_ORACLE_STALENESS_THRESHOLD_SEC
+        );
+    }
+
+    /// @dev We compute y value given a line defined by {x1,y1 and x2,y2} and x.
+    function _computeY(
+        uint256 x1,
+        uint256 y1,
+        uint256 x2,
+        uint256 y2,
+        uint256 x
+    ) private pure returns (uint256) {
+        // m = dlY/dlX
+        // c = y2 - m . x2
+        // y = m . x + c
+        int256 dlY = SafeCast.toInt256(y2) - SafeCast.toInt256(y1);
+        int256 dlX = SafeCast.toInt256(x2) - SafeCast.toInt256(x1);
+        int256 c = SafeCast.toInt256(y2) - ((SafeCast.toInt256(x2) * dlY) / dlX);
```

Comment:
I know this is reused from the BB, but looking at this again, is this simpler?
`return y1 + (x - x1) * dlY / dlX`